source/Calamari.Common/Features/Packages/ExtractionLimitsDecorator.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.IO;
+using Calamari.Common.Plumbing.Extensions;
+using Calamari.Common.Plumbing.Logging;
+using SharpCompress.Archives;
+using SharpCompress.Readers;
+
+namespace Calamari.Common.Features.Packages
+{
+    public class ExtractionLimitsDecorator : IPackageExtractor
```

Comment:
I love how you've used a decorator pattern to do this 👍 👍 

One question though, why is it called `ExtractionLimitsDecorator` when we are just logging metrics? I would've thought something along the lines of `ExtractionLoggingDecorator` would've made more sense. (and also the extension method)

Reply:
My next PR to enforce the limits makes it a better implementation too :)